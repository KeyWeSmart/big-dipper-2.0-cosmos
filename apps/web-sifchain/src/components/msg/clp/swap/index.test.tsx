import renderer from 'react-test-renderer';
import Swap from '@/components/msg/clp/swap';
import MsgSwap from '@/models/msg/clp/msg_swap';
import MockTheme from '@/tests/mocks/MockTheme';

// ==================================
// mocks
// ==================================
jest.mock('@/components/name', () => (props: JSX.IntrinsicElements['div']) => (
  <div id="Name" {...props} />
));

// ==================================
// unit tests
// ==================================
describe('screen: TransactionDetails/Swap', () => {
  it('matches snapshot', () => {
    const message: MsgSwap = {
      category: 'bank',
      type: 'MsgCreatePool',
      signer: 'signer',
      sentAsset: {
        symbol: 'udaric',
      },
      receivedAsset: {
        symbol: 'udaric',
      },
      sentAmount: '4000000',
      minReceivingAmount: '51000000',
      receivedAmount: '51000000',
      json: {},
    };

    const component = renderer.create(
      <MockTheme>
        <Swap message={message} />
      </MockTheme>
    );
    const tree = component?.toJSON();
    expect(tree).toMatchSnapshot();

    expect(component.root.findByProps({ 'data-testid': 'Trans' }).props.i18nKey).toEqual(
      'message_contents:MsgSwap'
    );
    expect(
      component.root.findByProps({ i18nKey: 'message_contents:MsgSwap' }).props.values
        .receivedAmount
    ).toEqual('51 DARIC');
  });

  afterEach(() => {
    jest.clearAllMocks();
  });
});
