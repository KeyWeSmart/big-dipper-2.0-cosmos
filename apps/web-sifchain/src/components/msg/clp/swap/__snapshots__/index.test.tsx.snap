// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`screen: TransactionDetails/Swap matches snapshot 1`] = `
.emotion-0 {
  margin: 0;
  font-size: 1rem;
  white-space: pre-wrap;
  letter-spacing: 0.5px;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 400;
  line-height: 1.5;
}

<p
  className="MuiTypography-root MuiTypography-body1 emotion-0"
>
  <span
    data-testid="Trans"
    i18nKey="message_contents:MsgSwap"
  >
    <div
      address="signer"
      id="Name"
      name="signer"
    />
    <b />
    {"sentAmount":"4 DARIC","receivedAmount":"51 DARIC"}
  </span>
</p>
`;
